chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -276,28 +277,72 @@ impl PresignatureManager {
             // To ensure there is no contention between different nodes we are only using triples
             // that we proposed. This way in a non-BFT environment we are guaranteed to never try
             // to use the same triple as any other node.
-            if let Some((triple0, triple1)) = triple_manager.take_two_mine().await {
+            if let Some((triple0, triple1)) = triple_manager.peek_two_mine() {
+                let id0 = triple0.id;
+                let id1 = triple1.id;
                 let presig_participants = active
                     .intersection(&[&triple0.public.participants, &triple1.public.participants]);
                 if presig_participants.len() < self.threshold {
+                    tracing::warn!(
+                        id0,
+                        id1,
+                        participants = ?presig_participants.keys_vec(),
+                        "running: participants are not above threshold for presignature generation"
+                    );
+                    return Ok(());
+                }
+
+                let state_views = presig_participants
+                    .iter()
+                    .filter_map(|(p, _)| Some((*p, state_views.get(p)?)));
+
+                // Filter out the active participants with the state views that have the triples we want to use.
+                let active_filtered = state_views
+                    .filter(|(_, state_view)| {
+                        if let StateView::Running {
+                            triple_postview, ..
+                        } = state_view
+                        {
+                            triple_postview.contains(&triple0.id)
+                                && triple_postview.contains(&triple1.id)
+                        } else {
+                            false
+                        }
+                    })
+                    .map(|(p, _)| p)
+                    .collect::<Vec<_>>();
+
+                if active_filtered.len() < self.threshold {
                     tracing::debug!(
```

Comment:
```suggestion
                    tracing::warning!(
```
Because this shouldn't really happen